cfme/storage/object_store_object.py

```diff
@@ -115,9 +116,14 @@ class ObjectStoreObjectCollection(BaseCollection):
     def all(self):
         """returning all Object Store Objects"""
         view = navigate_to(self, 'All')
-        view.toolbar.view_selector.select("Grid View")
-        objects = [self.instantiate(key=item)
-                   for item in view.entities.entity_names]
+
+        objects = []
+
+        # ToDo: use all_entity_names method as JS API issue (#2898) resolve.
```

Comment:
**REQUIRED:** This assumes that you will always want the provider filtered view and to return nothing, in the even that there is no provider filter. I don't think this is right. If there is no filter, we should return ALL of them. Otherwise, we shuldn't make the collection available on the appliance at all.

Reply:
@psav Its filtering objects as per provider. I am reading all objects and filter them with key attribute Cloud Provider. and return all objects of respective provider.

I used .filter() to provide provider in all() method as it's part of the collection class. all() method returning filtered object as per provider.

Maybe this wrong way of using .filter() method. please, can you suggest me exact use?

Reply:
Right, but the problem with the approach is that this is the ONLY supported way of using it. Meaning that if you don't supply the provider in the filter, you get an empty list.
It is expected that if you run a collection without any filters, that it should return ALL objects on the appliance.

Reply:
Adding exception so it will return ALL object without filter...